src/test/java/org/folio/logintest/LogoutTest.java

```diff
@@ -0,0 +1,143 @@
+        .setBaseUri("http://localhost:" + port)
+        .addHeader(XOkapiHeaders.URL, "http://localhost:" + mockPort)
+        .addHeader(XOkapiHeaders.TENANT, TENANT_DIKU)
+        .addHeader(XOkapiHeaders.TOKEN, "accesstoken")
+        .addHeader("Cookie", "accessToken=123; refreshToken=321")
+        .build();
+
+    specLogoutExpiredToken = new RequestSpecBuilder()
+        .setBaseUri("http://localhost:" + port)
+        .addHeader(XOkapiHeaders.URL, "http://localhost:" + mockPort)
+        .addHeader(XOkapiHeaders.TENANT, TENANT_DIKU)
+        .addHeader(XOkapiHeaders.TOKEN, "expiredtoken")
+        .addHeader("Cookie", "accessToken=expiredtoken; refreshToken=321")
+        .build();
+
+    specLogoutAll = new RequestSpecBuilder()
+        .setBaseUri("http://localhost:" + port)
+        .addHeader(XOkapiHeaders.URL, "http://localhost:" + mockPort)
+        .addHeader(XOkapiHeaders.TENANT, TENANT_DIKU)
+        .addHeader(XOkapiHeaders.TOKEN, "accesstoken")
+        .build();
+
+    specLogoutAllExpiredToken = new RequestSpecBuilder()
+        .setBaseUri("http://localhost:" + port)
+        .addHeader(XOkapiHeaders.URL, "http://localhost:" + mockPort)
+        .addHeader(XOkapiHeaders.TENANT, TENANT_DIKU)
+        .addHeader(XOkapiHeaders.TOKEN, "expiredtoken")
+        .build();
+
+    TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.1.0");
+    vertx.deployVerticle(Mocks.class.getName(), mockOptions)
+        .compose(res -> vertx.deployVerticle(RestVerticle.class.getName(), options))
+        .compose(res -> TestUtil.postSync(ta, TENANT_DIKU, port, vertx))
+        .onComplete(context.asyncAssertSuccess());
+  }
+
+  @Test
+  public void testLogout(final TestContext context) {
+      testLogoutCookieResponse(specLogout, LOGOUT_PATH);
+    }
+
+  @Test
+  public void testLogoutBadRequest(final TestContext context) {
+    RestAssured.given()
+      .spec(specLogoutExpiredToken)
+      .when()
+      .post(LOGOUT_PATH)
+      .then()
+      .log().all()
+      .statusCode(422);
+  }
+
+  @Test
+  public void testLogoutAll(final TestContext context) {
+    testLogoutCookieResponse(specLogoutAll, LOGOUT_ALL_PATH);
+  }
+
+  @Test
+  public void testLogoutBadRequestAll(final TestContext context) {
+    RestAssured.given()
+      .spec(specLogoutAllExpiredToken)
+      .when()
+      .post(LOGOUT_ALL_PATH)
+      .then()
+      .log().all()
+      .statusCode(422);
+  }
+
+  private void testLogoutCookieResponse(RequestSpecification spec, String path) {
+    RestAssured.given()
+        .spec(spec)
+        .when()
+        .post(path)
+        .then()
+        .log().all()
+        .statusCode(204)
+        .cookie(LoginAPI.REFRESH_TOKEN, RestAssuredMatchers.detailedCookie()
+            .value("")
+            .expiryDate(new Date(0))
+            .path("/authn"))
+        .cookie(LoginAPI.ACCESS_TOKEN, RestAssuredMatchers.detailedCookie()
+            .value("")
+            .expiryDate(new Date(0)));
```

Comment:
Please add

    .domain((String) null)  // not setting Domain disables subdomains
    .path((String) null)    // not setting Path allows all paths

Reply:
or

```
// not setting Domain disables subdomains: 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#domaindomain-value
.domain(is(nullValue()))
```

Setting a domain opens the cookie for all subdomains, for example hacker.okapi.folio.example.com. This is not what we want.

Not setting a domain restricts the cookie for the domain only, for example okapi.folio.example.com. This is what we want.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#domaindomain-value